rollout/experiment.go

```diff
@@ -64,8 +64,11 @@ func GetExperimentFromTemplate(r *v1alpha1.Rollout, stableRS, newRS *appsv1.Repl
 			Name:     templateStep.Name,
 			Replicas: templateStep.Replicas,
 		}
-		if templateStep.Weight != nil {
+		if templateStep.Weight != nil || templateStep.Service != nil {
 			template.Service = &v1alpha1.TemplateService{}
+			if templateStep.Service != nil && templateStep.Service.Name != "" {
```

Comment:
```suggestion
			if templateStep.Service.Name != "" {
```

you already checked for service not being nil

Reply:
This `templateStep.Service != nil` is for the case where templateStep.Weight != nil and templateStep.Service == nil, else `templateStep.Service.Name` will return an invalid memory address error

Reply:
but you already check it on line 67, so it can never be nil on line 69, right?

Reply:
Line 67 will pass even if Service is nil as long as Weight is not nil. Confirmed it by rerunning the unit tests as well